include/rdma/fabric.h

```diff
@@ -269,7 +269,8 @@ enum {
 	FI_PROTO_GNI,
 	FI_PROTO_RXM,
 	FI_PROTO_RXD,
-	FI_PROTO_MLX
+	FI_PROTO_MLX,
+	FI_PROTO_NETWORKDIRECT
```

Comment:
Unfortunately, I can't find any way to do this. As I see we can obtain Vendor Id and Device Id. For example, it may give us that Infiniband QDR of some vendor is used. 
Maybe I should ask someone from Microsoft ND team if there is such possibility. It is really good question and I should more deeper investigate it, because seems underlying protocol can be obtained somehow by ND serrvices.